core/src/main/scala/chisel3/internal/chisel-circt-binder/src/PanamaCIRCT.scala

```diff
@@ -0,0 +1,394 @@
+  private[circt] val sizeof = circt.MlirRegion.sizeof().toInt
+}
+object MlirRegion {
+  private[circt] def apply(ptr: MemorySegment) = new MlirRegion(ptr)
+}
+
+final case class MlirIdentifier(ptr: MemorySegment) extends ForeignType[MemorySegment] {
+  private[circt] def get = ptr
+  private[circt] val sizeof = circt.MlirIdentifier.sizeof().toInt
+}
+object MlirIdentifier {
+  private[circt] def apply(ptr: MemorySegment) = new MlirIdentifier(ptr)
+}
+
+final case class MlirLocation(ptr: MemorySegment) extends ForeignType[MemorySegment] {
+  private[circt] def get = ptr
+  private[circt] val sizeof = circt.MlirLocation.sizeof().toInt
+}
+object MlirLocation {
+  private[circt] def apply(ptr: MemorySegment) = new MlirLocation(ptr)
+}
+
+final case class MlirModule(ptr: MemorySegment) extends ForeignType[MemorySegment] {
+  private[circt] def get = ptr
+  private[circt] val sizeof = circt.MlirModule.sizeof().toInt
+}
+object MlirModule {
+  private[circt] def apply(ptr: MemorySegment) = new MlirModule(ptr)
+}
+
+final case class MlirOperation(ptr: MemorySegment) extends ForeignType[MemorySegment] {
+  private[circt] def get = ptr
+  private[circt] val sizeof = circt.MlirOperation.sizeof().toInt
+}
+object MlirOperation {
+  private[circt] def apply(ptr: MemorySegment) = new MlirOperation(ptr)
+}
+
+final case class MlirOperationState(ptr: MemorySegment) extends ForeignType[MemorySegment] {
+  private[circt] def get = ptr
+  private[circt] val sizeof = circt.MlirOperationState.sizeof().toInt
+}
+object MlirOperationState {
+  private[circt] def apply(ptr: MemorySegment) = new MlirOperationState(ptr)
+}
+
+final case class MlirType(ptr: MemorySegment) extends ForeignType[MemorySegment] {
+  private[circt] def get = ptr
+  private[circt] val sizeof = circt.MlirType.sizeof().toInt
+}
+object MlirType {
+  private[circt] def apply(ptr: MemorySegment) = new MlirType(ptr)
+}
+
+final case class MlirValue(ptr: MemorySegment) extends ForeignType[MemorySegment] {
+  private[circt] def get = ptr
+  private[circt] val sizeof = circt.MlirValue.sizeof().toInt
+}
+object MlirValue {
+  private[circt] def apply(ptr: MemorySegment) = new MlirValue(ptr)
+}
+
+final case class MlirStringRef(ptr: MemorySegment) extends ForeignType[MemorySegment] {
+  private[circt] def get = ptr
+  private[circt] val sizeof = circt.MlirStringRef.sizeof().toInt
+
+  override def toString: String = {
+    var slice = circt.MlirStringRef.data$get(ptr).asSlice(0, circt.MlirStringRef.length$get(ptr))
+    new String(slice.toArray(JAVA_BYTE))
+  }
+}
+object MlirStringRef {
+  private[circt] def apply(ptr: MemorySegment) = new MlirStringRef(ptr)
+}
+
+final case class FIRRTLBundleField(name: String, isFlip: Boolean, tpe: MlirType)
+
+//
+// MLIR & CIRCT Enums
+//
+
+sealed abstract class FIRRTLNameKind(val value: Int) extends ForeignType[Int] {
+  private[circt] def get = value
+  private[circt] val sizeof = 4 // FIXME: jextract doesn't export type for C enum
```

Comment:
How to fix this?

Reply:
The problem is `jextract` doesn't extract C enum to a Java class, so that we can't get the size of C enums.

We can simply assume the size of enums are 4, that should not be break easily from CIRCT unless someone explicitly changes the type of enumerations.